data/startup.sh

```diff
@@ -141,6 +141,84 @@ fetch_builtin_ctx() {
   echo "Done fetching builtin build context"
 }
 
+# need to forcefully stop journald service to unmount stateful partition.
+stop_journald_service() {
+  mkdir -p /etc/systemd/system/systemd-journald.service.d
+  cat > /etc/systemd/system/systemd-journald.service.d/override.conf<<EOF
+[Service]
+Restart=no
+EOF
+
+  /usr/bin/systemctl daemon-reload
+  /usr/bin/systemctl stop systemd-journald.socket
+  /usr/bin/systemctl stop systemd-journald-dev-log.socket
+  /usr/bin/systemctl stop systemd-journald-audit.socket
+  /usr/bin/systemctl stop syslog.socket
+  /usr/bin/systemctl stop systemd-journald.service
+}
+
+# this unit runs at shutdown time after everything but /tmp is unmounted
+create_run_after_unmount_unit(){
+  mkdir /tmp/last
+  mount -t tmpfs tmpfslast /tmp/last
+  # get OEMSize user input from meatadata
+  local -r oem_size="$(/usr/share/google/get_metadata_value \
+    attributes/OEMSize)"
+  cat > /etc/systemd/system/last-run.service<<EOF
+[Unit]
+Description=Run after everything unmounted
+DefaultDependencies=false
+Conflicts=shutdown.target
+Before=shutdown.target multi-user.target mnt-stateful_partition.mount var.mount mnt-disks.mount var-lib-docker.mount var-lib-toolbox.mount usr-share-oem.mount
+After=tmp-last.mount
+
+[Service]
+Type=oneshot
+RemainAfterExit=true
+ExecStart=/bin/true
+ExecStop=/tmp/last/extend-oem.bin /dev/sda 1 8 ${oem_size}
+TimeoutStopSec=600
+StandardOutput=tty
+StandardError=tty
+TTYPath=/dev/ttyS2
+EOF
+}
+
+extend_oem_partition(){
+  echo "Checking whether need to extend OEM partition..."
+
+  # get user input from meatadata
+  local -r oem_size="$(/usr/share/google/get_metadata_value \
+    attributes/OEMSize)"
+
+  if [ ${oem_size} != "" ]; then 
```

Comment:
nit: In bash, use '[[' instead of '['. Also, you can check if a variable is empty with "[[ -z "${oem_size}" ]]". You can check if a variable is non-empty with "[[ -n "${oem_size}" ]]".

Also, it's better to quote variables to prevent shell wildcard expansion: "${oem_size}".